synapse/config/sso.py

```diff
@@ -44,6 +48,16 @@ def generate_config_section(self, **kwargs):
         # Additional settings to use with single-sign on systems such as SAML2 and CAS.
         #
         sso:
+            # Uncomment the following to disable the extra confirmation step before
+            # redirecting to the client after a successful authentication. This step is
+            # enabled by default.
+            #
+            # Make sure the SSO backend allows sensible authentication options (e.g. no
+            # "passwordless" authentication) before disabling this step, otherwise users
+            # could be vulnerable to phishing attacks.
```

Comment:
Could you expand this a bit to highlight when its safe and not safe to disable this, e.g.

> Authentication methods that involve sending an authentication link to an email which includes a redirect URL are vulnerable to "open redirect" phishing attacks, and in such cases it is highly recommended to keep this option enabled.

Or something